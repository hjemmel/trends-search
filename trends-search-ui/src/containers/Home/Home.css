.home {
    display: flex;
    flex: 1;
}

.main {
    flex: 3;
}

.options {
    flex: 1;
    background-color: antiquewhite;
}
