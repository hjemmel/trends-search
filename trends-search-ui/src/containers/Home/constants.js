export const LOAD_TRENDS = 'trends/Home/LOAD_TRENDS';
export const LOAD_TRENDS_SUCCESS = 'trends/Home/LOAD_TRENDS_SUCCESS';
export const LOAD_TRENDS_ERROR = 'trends/Home/LOAD_TRENDS_ERROR';
