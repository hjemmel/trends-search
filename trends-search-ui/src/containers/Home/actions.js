import {LOAD_TRENDS, LOAD_TRENDS_ERROR, LOAD_TRENDS_SUCCESS} from './constants';

export function loadTrends () {
  return {
    type: LOAD_TRENDS
  };
}

export function trendsLoaded (trends) {
  return {
    type: LOAD_TRENDS_SUCCESS,
    trends
  };
}

export function trendsLoadingError (error) {
  return {
    type: LOAD_TRENDS_ERROR,
    error
  };
}
