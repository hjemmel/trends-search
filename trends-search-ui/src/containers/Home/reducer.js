import {LOAD_TRENDS, LOAD_TRENDS_ERROR, LOAD_TRENDS_SUCCESS} from './constants';
import data from './data';

const initialState = {
  loading: false,
  error: false,
  trends: []
};

const trendsReducer = (state = initialState, action) => {
  switch (action.type) {
    case LOAD_TRENDS:
      return {
        ...state,
        loading: false,
        trends: data
      };
    case LOAD_TRENDS_SUCCESS:
      return {
        ...state,
        loading: false,
        trends: data
      };
    case LOAD_TRENDS_ERROR:
      return {
        ...state,
        error: true
      };
    default:
      return state;
  }
};

export default trendsReducer;
