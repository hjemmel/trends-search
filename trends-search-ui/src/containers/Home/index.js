import React from 'react';
import './Home.css';
import BarChart from '../../components/BarChart';
import {connect} from 'react-redux';
import {loadTrends} from './actions';
import PropTypes from 'prop-types';

class Home extends React.Component {
  componentDidMount () {
    const {loadTrends} = this.props;
    loadTrends();
  }

  render (props) {
    return (
      <div className='home'>
        <div className='main'>
          <BarChart data={this.props.trends} />
        </div>
        <div className='options' />
      </div>);
  };
}

const mapStateToProps = state => {
  return {
    trends: state.home.trends
  };
};

const mapDispatchToProps = ({
  loadTrends: loadTrends
});

Home.propTypes = {
  loadTrends: PropTypes.func,
  trends: PropTypes.arrayOf(PropTypes.shape({
    'country': PropTypes.string,
    'hot dog': PropTypes.number,
    'hot dogColor': PropTypes.string,
    'burger': PropTypes.number,
    'burgerColor': PropTypes.string,
    'sandwich': PropTypes.number,
    'sandwichColor': PropTypes.string,
    'kebab': PropTypes.number,
    'kebabColor': PropTypes.string,
    'fries': PropTypes.number,
    'friesColor': PropTypes.string,
    'donut': PropTypes.number,
    'donutColor': PropTypes.string
  }))
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
