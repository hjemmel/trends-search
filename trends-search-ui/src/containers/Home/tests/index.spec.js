import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Home from '../index';

describe('<Home />', () => {
  it('should render properly', () => {
    const renderedComponent = shallow(<Home />);
    expect(toJson(renderedComponent)).toMatchSnapshot();
  });

  it('should render the header', () => {
    const renderedComponent = shallow(<Home />);
    expect(renderedComponent.contains('hans')).toBeTruthy();
  });
});
