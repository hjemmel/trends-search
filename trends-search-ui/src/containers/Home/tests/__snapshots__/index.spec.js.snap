// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render properly 1`] = `
<div
  className="home"
>
  <div
    className="main"
  >
    hans
  </div>
  <div
    className="options"
  />
</div>
`;
